cmd/vsphere-xcopy-volume-populator/internal/populator/remote_esxcli.go

```diff
@@ -79,8 +79,12 @@ func (p *RemoteEsxcliPopulator) Populate(sourceVMDKFile string, volumeHandle str
 	}
 	klog.Infof("Got ESXI host: %s", host)
 
+	err = ensureVib(p.VSphereClient, host, vmDisk.Datastore, vibVersion)
+	if err != nil {
+		return fmt.Errorf("failed to ensure VIB is installed: %w", err)
```

Comment:
Do we want to fail fail when we fail this step?
What if the VIB is present but we can't upload to it because of the certs/permissions/networking?

Reply:
if the vib is already present we don't upload it again , unless the version changed. 
So this is once per ESX, per vib version